### PR TITLE
Modify overlay and add driver for Chipdip DAC

### PR DESCRIPTION
**Pull request by chipdipru**
The overlay has been changed and driver has been added for Chipdip DAC, tested on Raspberry PI3.

### CODE REVIEW

File: sound/soc/bcm/chipdip-dac.c [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+static int current_width = 0;
+static int current_rate = 0;
+
+static void snd_rpi_chipdip_dac_gpio_array_set(int value);
+static void snd_rpi_chipdip_dac_gpio_set(struct gpio_desc *gpio_item, int value);
+
+static void snd_rpi_chipdip_dac_gpio_array_set(int value)
+{
+        int i = 0;
+        for (i = 0; i < HW_PARAMS_GPIO_COUNT; i++)
```
Comment (pelwell) [inline sound/soc/bcm/chipdip-dac.c]: Kernel coding rules say there should be a blank line between any variable declarations and the first statement.

File: sound/soc/bcm/chipdip-dac.c [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+
+static int snd_rpi_chipdip_dac_init(struct snd_soc_pcm_runtime *rtd)
+{	
+	return 0;
+}
+
+static int snd_rpi_chipdip_dac_hw_params(struct snd_pcm_substream *substream, struct snd_pcm_hw_params *params)
+{
+	int ret = 0;
+        int gpio_change_pending = 0;
```
Comment (pelwell) [inline sound/soc/bcm/chipdip-dac.c]: This line (like many others) uses spaces for indentation when it should use TABs.

**Review by pelwell — COMMENTED** [review]
There are some cosmetic (whitespace) issues, but basically this looks fine. Please fix the whitespace, then squash the changes to make three commits:
1. The driver and its Makefile and Kconfig changes.
2. The overlay and its Makefile and README changes.
3. The updated _defconfig files with "SND_BCM2708_SOC_CHIPDIP_DAC=m" (you can grab this from here: https://github.com/pelwell/linux/commit/24767719b09ca526c9df941be8fe99910de85786)

**Comment by chipdipru** [comment]
Phil, thanks for the review, corrections have been made.

**Comment by pelwell** [comment]
The driver C file has ended up with DOS line endings (CR+LF), whereas Linux line endings should be just LF. Also, delete any trailing whitespace.

Once you've done that, squash the commits down to the three I described above.

**Comment by pelwell** [comment]
I've fixed up the commits for you. Write three commit messages here in code sections - like this:
```
First commit message

This is the first commit.

Signed-off-by: User Name <user@email.com>
```
I can graft them on and merge the PR.

**Comment by chipdipru** [comment]
Thank you. I am not sure I understand what I shall do. Please correct me if something is wrong.

1)
`Update and rename chipdip-i2s-master-dac-overlay`

`Simple card fragment removed, GPIO fragment added.`

`Signed-off-by: Evgenij Sapunov <evgenij.sapunov@chipdip.ru>`


2)
`Add chipdip-dac driver`

`Driver chipdip-dac.c added into sound/soc/bcm/,
files sound/soc/bcm/Kconfig and sound/soc/bcm/Makefile
updated`

`Signed-off-by: Evgenij Sapunov <evgenij.sapunov@chipdip.ru>`


3)
`Update _defconfig files`

`Files arch/arm/configs/bcm2709_defconfig,
arch/arm/configs/bcm2711_defconfig,
arch/arm/configs/bcmrpi_defconfig,
arch/arm64/configs/bcm2711_defconfig,
arch/arm64/configs/bcmrpi3_defconfig
updated.`

`Signed-off-by: Evgenij Sapunov <evgenij.sapunov@chipdip.ru>`

**Comment by pelwell** [comment]
That's perfect.

I'm guessing that either you are using the GitHub GUI for creating your pull requests or you haven't discovered `git rebase -i`. I recommend taking the time to learn how to edit and squash commits, then using `git push -f` to update the branch you created your PR from until it looks correct - it will save you a lot of time in the future.

**Comment by chipdipru** [comment]
Thank you!

**Pull request merged**